packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationBackgroundService.kt

```diff
@@ -84,7 +84,7 @@ class PushNotificationBackgroundService : JobIntentService(), MethodChannel.Meth
         }
     }
 
-    fun createAndRunBackgroundEngine(context: Context, loader: FlutterLoader ){
+    fun createAndRunBackgroundEngine(context: Context, loader: FlutterLoader, mockEngine: FlutterEngine? ){
```

Comment:
Okay, makes sense you need to mock the FlutterEngine, what about rewriting the code to be more clear as mentioned above: 

`backgroundFlutterEngine = mockEngine ?: FlutterEngineCache....`

Reply:
What are you testing inside the if statement?  It seems you're just putting the engine within the FlutterEngineCache?

Since as you say you can mock the FlutterEngineCache, why not have that mock return the mockEngine and avoid the if statement?

Reply:
I know there is no logic to test inside the if statement, I was more trying to cover that since it's an important if block for the functioning of killed state. 

I do return a mockEngine and skip the if block in [another test](https://github.com/aws-amplify/amplify-flutter/blob/fix/push-android-tests/packages/notifications/push/amplify_push_notifications/android/src/test/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationBackgroundServiceTest.kt#L91).